extendr-api/src/robj_ndarray.rs

```diff
@@ -94,138 +94,146 @@ where
     }
 }
 
-#[test]
-fn test_from_robj() {
-    test! {
-        assert_eq!(
-            <ArrayView1<f64>>::from_robj(&Robj::from(1.)),
-            Ok(ArrayView1::<f64>::from(&[1.][..]))
-        );
-        assert_eq!(
-            <ArrayView1<i32>>::from_robj(&Robj::from(1)),
-            Ok(ArrayView1::<i32>::from(&[1][..]))
-        );
-        assert_eq!(
-            <ArrayView1<Rbool>>::from_robj(&Robj::from(true)),
-            Ok(ArrayView1::<Rbool>::from(&[TRUE][..]))
-        );
-
-        let robj = R!("matrix(c(1, 2, 3, 4, 5, 6, 7, 8), ncol=2, nrow=4)")?;
-        let mx = <ArrayView2<f64>>::from_robj(&robj)?;
-        assert_eq!(mx[[0, 0]], 1.0);
-        assert_eq!(mx[[1, 0]], 2.0);
-        assert_eq!(mx[[2, 0]], 3.0);
-        assert_eq!(mx[[3, 0]], 4.0);
-        assert_eq!(mx[[0, 1]], 5.0);
-        assert_eq!(mx[[1, 1]], 6.0);
-        assert_eq!(mx[[2, 1]], 7.0);
-        assert_eq!(mx[[3, 1]], 8.0);
-
-        // check basic logic of fortran-order matrices.
-        let col0 = mx.column(0);
-        assert_eq!(col0[0], 1.0);
-        assert_eq!(col0[1], 2.0);
-        assert_eq!(col0[2], 3.0);
-        assert_eq!(col0[3], 4.0);
-
-        // check integer matrices
-        let robj = R!("matrix(c(1L, 2L, 3L, 4L, 5L, 6L, 7L, 8L), ncol=2, nrow=4)")?;
-        let mx = <ArrayView2<i32>>::from_robj(&robj)?;
-        assert_eq!(mx[[0, 0]], 1);
-        assert_eq!(mx[[1, 0]], 2);
-        assert_eq!(mx[[2, 0]], 3);
-        assert_eq!(mx[[3, 0]], 4);
-        assert_eq!(mx[[0, 1]], 5);
-        assert_eq!(mx[[1, 1]], 6);
-        assert_eq!(mx[[2, 1]], 7);
-        assert_eq!(mx[[3, 1]], 8);
-
-        // check logical matrices
-        let robj = R!("matrix(c(T, T, T, T, F, F, F, F), ncol=2, nrow=4)")?;
-        let mx = <ArrayView2<Rbool>>::from_robj(&robj)?;
-        assert_eq!(mx[[0, 0]], TRUE);
-        assert_eq!(mx[[1, 0]], TRUE);
-        assert_eq!(mx[[2, 0]], TRUE);
-        assert_eq!(mx[[3, 0]], TRUE);
-        assert_eq!(mx[[0, 1]], FALSE);
-        assert_eq!(mx[[1, 1]], FALSE);
-        assert_eq!(mx[[2, 1]], FALSE);
-        assert_eq!(mx[[3, 1]], FALSE);
-
-        // check raw matrices
-        // let robj = r!(Matrix::new(vec![1_u8, 2, 3, 4, 5, 6, 7, 8], 4, 2));
-        // let mx = <ArrayView2<u8>>::from_robj(&robj)?;
-        // assert_eq!(mx[[0, 0]], 1);
-        // assert_eq!(mx[[1, 0]], 2);
-        // assert_eq!(mx[[2, 0]], 3);
-        // assert_eq!(mx[[3, 0]], 4);
-        // assert_eq!(mx[[0, 1]], 5);
-        // assert_eq!(mx[[1, 1]], 6);
-        // assert_eq!(mx[[2, 1]], 7);
-        // assert_eq!(mx[[3, 1]], 8);
+#[cfg(test)]
+mod test {
+    use super::*;
+    use rstest::rstest;
+
+    #[rstest]
+    fn test_from_robj() {
```

Comment:
Any idea of how to switch thist test to `rstest`?

Reply:
Not sure what you mean. This one was changed to `rstest`?

Reply:
Sorry I did not explain myself. Can we use a fixture to provide test case data for this test as well, right now it is a lot of copy-paste.